src/tests/JIT/SIMD/ShiftOperations.cs

```diff
@@ -9,19 +9,76 @@
 public class Test
 {
     [MethodImpl(MethodImplOptions.NoInlining)]
-    private static ulong Shlx64bit(ulong x, int y) => x << y;
+    private static R Shlx64bit<T, R>(T x, int y)
+    {
+        switch (x)
+        {
+            case ulong a:
+                ulong resUlong = ((ulong)a) << y;
```

Comment:
Why doesn't this generic work?

```c#
T res = ((T)x) << y;
return (R)Convert.ChangeType(res,typeof(R));
```

Reply:
shift operators does not work on generics. 
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/bitwise-and-shift-operators
`Because the shift operators are defined only for the int, uint, long, and ulong types, the result of an operation always contains at least 32 bits. If the left-hand operand is of another integral type (sbyte, byte, short, ushort, or char), its value is converted to the int type`

Reply:
@kunalspathak, this is the error message.
`error CS0019: Operator '<<' cannot be applied to operands of type 'T' and 'int'`
All tests passed, so merging it now. Thanks for all the code reviews.